src/test/java/org/broadinstitute/barclay/argparser/CollectionArgumentUnitTests.java

```diff
@@ -0,0 +1,306 @@
+        // provided for that argument, since even though its redundant, its harmless and would allow a script or
+        // workflow to use the same command line with either parser behavior.
+        // TODO: We should probably still throw if the null appears AFTER some other value for that argument.
+        // TODO: Add a test for this if we do it.
+        doSetCollectionValues(Collections.<CommandLineParserOptions>emptySet(), args);
+    }
+
+    // Test using null to clear a list and then set values, append mode
+    @Test(dataProvider="nullAndAddArgs")
+    public void testAppendModeNullAndSetValues(final String[] args) {
+        CollectionWithDefaultValues o = doSetCollectionValues(
+                Collections.singleton(CommandLineParserOptions.APPEND_TO_COLLECTIONS),
+                args
+        );
+        // specifying null *AND* other arguments in APPEND mode is the equivalent of replace,
+        // so verify that we've cleared the default values and added the new ones
+        Assert.assertEquals(o.LIST, makeList("baz", "frob"));
+    }
+
+    // Test that replace mode retains values given no args
+    @Test
+    public void testReplaceModeRetainDefaultValues() {
+        CollectionWithDefaultValues o = doSetCollectionValues(
+                Collections.<CommandLineParserOptions>emptySet(),
+                new String[]{}
+        );
+        Assert.assertEquals(o.LIST, makeList("foo", "bar"));
+    }
+
+    // Test that append mode retains values given no args
+    @Test
+    public void testAppendModeRetainDefaultValues() {
+        CollectionWithDefaultValues o = doSetCollectionValues(
+                Collections.singleton(CommandLineParserOptions.APPEND_TO_COLLECTIONS),
+                new String[]{}
+        );
+        Assert.assertEquals(o.LIST, makeList("foo", "bar"));
+    }
+
+    //////////////////////////////////////////////////////////////////
+    // tests for .list files
+
+    class CollectionForListFileArguments {
+        @Argument
+        public List<String> LIST = makeList("foo", "bar");
+
+        @Argument
+        public List<String> LIST2 = makeList("baz");
+    }
+
+    @DataProvider(name="listFileArguments")
+    public Object[][] listFileArguments() {
+        return new Object[][] {
+                {
+                        // replace mode
+                        Collections.EMPTY_SET,                                                  // parser options
+                        new String[] {"shmiggle0", "shmiggle1", "shmiggle2"},                   // args
+                        new String[] {"shmiggle0", "shmiggle1", "shmiggle2"},                   // expected result
+                },
+                {
+                        // append mode
+                        Collections.singleton(CommandLineParserOptions.APPEND_TO_COLLECTIONS),  // parser options
+                        new String[] {"shmiggle0", "shmiggle1", "shmiggle2"},                   // args
```

Comment:
same thing as above, i'd probably extract a variable for the repeated  declarations